### PR TITLE
Pvls fixes

### PR DESCRIPTION
**Pull request by ningosi**
Including correct encounter types in the denominator and those patients on ART for at least 3 months

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java [hunk 1]
```diff
@@ -67,24 +67,32 @@ public static String getPatientsHavingViralLoadInLast12Months(
...
-            fsrEncounter,
-            vlConceptQuestion,
-            vlQualitativeConceptQuestion);
-
-    return sql;
+    String query =
+        "SELECT p.patient_id FROM  patient p INNER JOIN encounter e ON p.patient_id=e.patient_id INNER JOIN"
+            + " obs o ON e.encounter_id=o.encounter_id "
+            + " WHERE p.voided=0 AND e.voided=0 AND o.voided=0 AND "
+            + " e.encounter_type IN (%d,%d,%d,%d,%d) AND "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java:94]: I believe here should be 4 params??

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java [hunk 2]
```diff
@@ -67,24 +67,32 @@ public static String getPatientsHavingViralLoadInLast12Months(
...
+            + " WHERE p.voided=0 AND e.voided=0 AND o.voided=0 AND "
+            + " e.encounter_type IN (%d) AND o.concept_id=%d AND o.value_numeric IS NOT NULL AND "
+            + " o.obs_datetime BETWEEN date_add(date_add(:endDate, interval -12 MONTH), interval 1 day) AND :endDate AND "
+            + " e.location_id=:location ";
+    return String.format(
+        query,
+        labEncounter,
+        adultSeguimentoEncounter,
+        pediatriaSeguimentoEncounter,
+        mastercardEncounter,
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java:110]: Which mean this guy should not be here

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java [hunk 3]
```diff
@@ -50,7 +58,8 @@ public static String getPatientsWithViralLoadSuppression(
         mastercardEncounter,
         fsrEncounter,
         vlConceptQuestion,
-        vlConceptQuestion);
+        mastercardEncounter,
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java:69]: Some params are missing here

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java [hunk 4]
```diff
@@ -40,7 +41,14 @@ public static String getPatientsWithViralLoadSuppression(
             + " INNER JOIN obs ON obs.person_id=ultima_carga.patient_id AND obs.obs_datetime="
             + "ultima_carga.data_carga  WHERE obs.voided=0 AND obs.concept_id=%d "
             + " AND obs.location_id=:location AND"
-            + " obs.value_numeric < 1000";
+            + " obs.value_numeric < 1000 "
+            + " UNION "
+            + " SELECT p.patient_id FROM  patient p INNER JOIN encounter e ON p.patient_id=e.patient_id INNER JOIN "
+            + " obs o ON e.encounter_id=o.encounter_id "
+            + " WHERE p.voided=0 AND e.voided=0 AND o.voided=0 AND "
+            + " e.encounter_type IN (%d) AND o.concept_id=%d AND o.value_numeric IS NOT NULL AND "
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java:49]: The concept 1305 is missing

**Review by psmatsinhe — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java (hunk 1 above)
Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java:94]: fixed

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java (hunk 3 above)
Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java:69]: fixed

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java (hunk 4 above)
Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/library/queries/ViralLoadQueries.java:49]: done

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForMoreThanXmonthsCalcultion.java [hunk 5]
```diff
@@ -62,23 +64,43 @@ public CalculationResultMap evaluate(
...
-          isOnArtForMoreThan3Months = true;
+      Date artStartDate;
+      if (artStartDateResult != null) {
+        artStartDate = (Date) artStartDateResult.getValue();
+        Obs lastVlObs = EptsCalculationUtils.resultForPatient(lastVl, ptId);
+        Obs lastQualitativeResults =
+            EptsCalculationUtils.resultForPatient(qViralLoadResultsMap, ptId);
+        if (checkNotNull(artStartDateResult, lastVlObs)) {
+
+          Date lastVlDate = lastVlObs.getObsDatetime();
```
Comment (psmatsinhe) [inline api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForMoreThanXmonthsCalcultion.java]: Think we should also check if the lastVlDate is not null, I believe that will not make much effect  but as we checked on line 96 we should do the same!

**Review by psmatsinhe — APPROVED** [review]

File: api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForMoreThanXmonthsCalcultion.java (hunk 5 above)
Comment (ningosi) [reply api/src/main/java/org/openmrs/module/eptsreports/reporting/calculation/pvls/OnArtForMoreThanXmonthsCalcultion.java]: done

**Pull request merged**